routes/ci.js

```diff
@@ -116,15 +116,14 @@ router.post('/register', authMiddleware.isLoggedIn, function (req, res, next) {
 
 router.post('/webhook', function(req, res, next) {
   var event = req.get('X-GitHub-Event');
-  var branch = req.body.ref.split("/")[2];
+  var branch;
   var repositoryName = req.body.repository.full_name;
-  if (branch === "gh-pages") {
-    return res.json({
-      status: false,
-      description: "No operation on pushes to gh-pages branch"
-    });
-  }
   var query = {};
+  if (event === "pull_request") {
+    branch = req.body.pull_request.base.ref;
+  } else {
```

Comment:
@imujjwal96 I have check your suggestion. when I'm splitting undefined route is freezing.